pkg/apis/rollouts/v1alpha1/experiment_types.go

```diff
@@ -67,6 +74,8 @@ type TemplateSpec struct {
 	Selector *metav1.LabelSelector `json:"selector" protobuf:"bytes,4,opt,name=selector"`
 	// Template describes the pods that will be created.
 	Template corev1.PodTemplateSpec `json:"template" protobuf:"bytes,5,opt,name=template"`
+	// CreateService determines if a service should be created for the template
+	CreateService bool `json:"createService,omitempty" protobuf:"varint,6,opt,name=createService"`
```

Comment:
I think a boolean might be short-sighted in case we need knobs to control **_how_** services should be generated (e.g. with a specific name) or possibly referenced and modified (like blue-green activeService).

Instead of a simple boolean, I think we can use an empty struct to indicate it should be created:

```yaml
spec:
  templates:
  - name: foo
    service: {}
```

Then in the future (not as a part of this PR), we could introduce:

```yaml
spec:
  templates:
  - name: foo
    service:
      name: foo
```

or:

```yaml
spec:
  templates:
  - name: foo
    service:
      generateName: foo-
```